code/modules/chemistry/tools/food_and_drink.dm

```diff
@@ -1726,5 +1737,7 @@
 			src.reagents.inert = 0
 			src.reagents.handle_reactions()
 			src.reagents.inert = 1
+			if ((user.mind.assigned_role == "Barman") && (prob(5)))
+				JOB_XP(user, "Bartender", 2)
```

Comment:
ideally this would give XP based on # of reactions that occurred due to the shake, or at least, make sure that at least 1 reaction happened. idk if there's an easy way to check that though